### PR TITLE
New enum + constructor added for `nullSafetyMigrationResult`

### PR DESCRIPTION
**Pull request by eliasyishak**
Related to tracker issue:
- https://github.com/flutter/flutter/issues/128251

This event is for the [`NullSafetyAnalysisEvent`](https://github.com/flutter/flutter/blob/master/packages/flutter_tools/lib/src/reporting/events.dart#L256-L315) class.

This event has a more involved `send` method that actually [sends 3 separate events](https://github.com/flutter/flutter/blob/master/packages/flutter_tools/lib/src/reporting/events.dart#L293-L301). I have taken those 3 separate events and consolidated them down to just one event with 4 parameters included.

---

- [x] I’ve reviewed the contributor guide and applied the relevant portions to this PR.

<details>
  <summary>Contribution guidelines:</summary><br>

- See our [contributor guide](https://github.com/dart-lang/.github/blob/main/CONTRIBUTING.md) for general expectations for PRs.
- Larger or significant changes should be discussed in an issue before creating a PR.
- Contributions to our repos should follow the [Dart style guide](https://dart.dev/guides/language/effective-dart) and use `dart format`.
- Most changes should add an entry to the changelog and may need to [rev the pubspec package version](https://github.com/dart-lang/sdk/wiki/External-Package-Maintenance#making-a-change).
- Changes to packages require [corresponding tests](https://github.com/dart-lang/.github/blob/main/CONTRIBUTING.md#Testing).

Note that many Dart repos have a weekly cadence for reviewing PRs - please allow for some latency before initial review feedback.
</details>

### CODE REVIEW

File: pkgs/unified_analytics/lib/src/event.dart [hunk 1]
```diff
@@ -43,6 +43,30 @@ final class Event {
           'method': method,
         };
 
+  /// Contains information about null safety migration within a flutter project.
```
[the remark this answers is not in the document]
Comment (eliasyishak) [reply pkgs/unified_analytics/lib/src/event.dart:46]: That's a good question, from what I understand looking at the code, it doesn't seem like we are using any external tool within flutter-tools

https://github.com/flutter/flutter/blob/master/packages/flutter_tools/lib/src/reporting/events.dart#L276-L302

It looks like it has its own logic for calculating the number of migrated packages by checking attributes on `LanguageVersion`

[the remark this answers is not in the document]
Comment (bwilkerson) [reply pkgs/unified_analytics/lib/src/event.dart:46]: Got it. This is for measuring how much of the code being used with a given version of Flutter has been migrated, and has nothing to do with the migration tool. Thanks!

(The rename suggestion still stands, but that's completely up to you.)

[the remark this answers is not in the document]
Comment (eliasyishak) [reply pkgs/unified_analytics/lib/src/event.dart:46]: Renaming it makes sense to me, anything to avoid confusion!

**Comment by christopherfujino** [comment]
I actually don't think we need this event anymore. I deleted this event in https://github.com/flutter/flutter/pull/123031/files#diff-aee5c7b97c66ed433071dcf682d8643bf6341749f7b14616227f136373dc7363L257, however that got reverted only because we still support non-null-safe code internally.

**Comment by eliasyishak** [comment]
> I actually don't think we need this event anymore.

Sounds good, I'll make a note that we are deprecating this event in the tracker issue and close

**Pull request closed without merging**